### PR TITLE
Merge

### PR DESCRIPTION
**Pull request by kbodner**
I added several new bacterial segmentation functions to subdetect, included a note about efficient usage of apply, dealt with numpy bugs in preprocess operation and utils.

### CODE REVIEW

File: celltk/utils/preprocess_utils.py [hunk 1]
```diff
@@ -79,12 +79,13 @@ def curvature_anisotropic_smooth(img, NUMITER=10):
...
 
 def remove_odd_addback(img, func):
-    rshape, cshape = img.shape
+    rshape, cshape, zshape = img.shape #changed by KB to be compatible with a numpy update issue
     if not rshape % 2 == 0:
         img = img[:-1, :]
     if not cshape % 2 == 0:
         img = img[:, :-1]
-    img = func(img)
+    img=img[:,:,0] # added by KB to be compatible with a numpy update issue
```
Comment (prismofeverything) [inline celltk/utils/preprocess_utils.py]: Formatting is a little funny here? (the `=` needs spaces)

File: celltk/preprocess_operation.py [hunk 2]
```diff
@@ -133,6 +155,7 @@ def flatfield_references(img, ff_paths=['Pos0/img00.tif', 'Pos1/img01.tif'], exp
     for path in store:
         ff_store.append(imread(path))
     ff = np.median(np.dstack(ff_store), axis=2)
+    ff = np.squeeze(ff) # added by Katie to fix bug in new version of numpy
```
Comment (prismofeverything) [inline celltk/preprocess_operation.py]: Which version of numpy are you using @kbodner? Is it the same one @braysia is using? Just want to make sure we aren't introducing incompatibilities in the dependencies.

Comment (kbodner) [reply celltk/preprocess_operation.py]: v1.13.3. I believe these issues started after Jerry performed an update on numpy.

Comment (prismofeverything) [reply celltk/preprocess_operation.py]: Ah okay, so we are all on the same version then. Sounds good, approved from me.

Comment (braysia) [reply celltk/preprocess_operation.py]: I'm using v1.14.6 and I couldn't reproduce the issue. I guess this suggests the way you pass ff_paths might not be correct - I'm happy to see the input file if you have one.

File: celltk/subdetect_operation.py [hunk 3]
```diff
@@ -184,8 +185,56 @@ def cytoplasm_levelset(labels, img, niter=20, dt=-0.5, thres=0.5):
...
+    """
+   labels = constant_thres(img, THRES=THRES)
+   if labels.any():
+        labels, comb, nuc_prop, nuc_loc = label_nearest(img, labels, nuc,dist)
+   from skimage.morphology import remove_small_objects
+   labels = remove_small_objects(labels, MINAREA)
+   return labels.astype(np.uint16)
+
+def segment_bacteria_return_cyto_no_bac(nuc, img, slen=3, SIGMA=0.5,THRES=20, CLOSE=20, THRESCHANGE=1000, MINAREA=5, dist=25):
+   """ Segment bacteria and assign to closest nucleus and return the mask back without regions containing bacteria
```
Comment (braysia) [inline celltk/subdetect_operation.py]: Please modify your tab behavior to 4x space.

File: celltk/preprocess_operation.py (hunk 2 above)
Comment (braysia) [reply celltk/preprocess_operation.py]: @prismofeverything
All the changes in existing functions should be discarded as those errors seems to be expected behaviors.
I just want to accept the addition of functions. What do you think the easy way to go forward? Re-send pull requests?

**Comment by prismofeverything** [comment]
Awesome, I think the last thing to do here is fix those funny permissions on the unchanged files. @kbodner do you want to do that? @braysia If you give me write access to this repo I could do it as well, let me know. Hoping to get started on the test deploy to GC for CellTK this week.

Thanks for doing all this!

**Comment by kbodner** [comment]
@prismofeverything  All permissions should now be fixed!

File: celltk/preprocess_operation.py (hunk 2 above)
Comment (kbodner) [reply celltk/preprocess_operation.py]: done. I deleted the extra lines I added to fix perceived numpy bugs

Comment (prismofeverything) [reply celltk/preprocess_operation.py]: @kbodner Awesome! Once you push to this branch we should be ready to merge then.

**Pull request merged**